--- a/examples/svc_encoder_rtc.cc
+++ b/examples/svc_encoder_rtc.cc
@@ -135,11 +135,16 @@
 #if CONFIG_AV1_HIGHBITDEPTH
 static const struct arg_enum_list bitdepth_enum[] = { { "8", AOM_BITS_8 },
                                                       { "10", AOM_BITS_10 },
+                                                      { "12", AOM_BITS_12 },
                                                       { NULL, 0 } };
 
 static const arg_def_t bitdepth_arg = ARG_DEF_ENUM(
     "d", "bit-depth", 1, "Bit depth for codec 8 or 10. ", bitdepth_enum);
 #endif  // CONFIG_AV1_HIGHBITDEPTH
+static const arg_def_t profile_arg =
+    ARG_DEF(NULL, "profile", 1, "Profile, after --bit-depth");
+static const arg_def_t monochrome_arg =
+    ARG_DEF(NULL, "monochrome", 0, "Code luma only");
 
 static const arg_def_t *svc_args[] = {
   &frames_arg,
@@ -160,6 +165,8 @@
 #if CONFIG_AV1_HIGHBITDEPTH
   &bitdepth_arg,
 #endif
+  &profile_arg,
+  &monochrome_arg,
   &speed_arg,
   &bitrates_arg,
   &dropframe_thresh_arg,
@@ -400,10 +407,18 @@
           enc_cfg->g_input_bit_depth = 10;
           enc_cfg->g_profile = 0;
           break;
+        case AOM_BITS_12:
+          enc_cfg->g_input_bit_depth = 12;
+          enc_cfg->g_profile = 2;
+          break;
         default:
           die("Error: Invalid bit depth selected (%d)\n", enc_cfg->g_bit_depth);
       }
 #endif  // CONFIG_VP9_HIGHBITDEPTH
+    } else if (arg_match(&arg, &profile_arg, argi)) {
+      enc_cfg->g_profile = arg_parse_uint(&arg);
+    } else if (arg_match(&arg, &monochrome_arg, argi)) {
+      enc_cfg->monochrome = 1;
     } else if (arg_match(&arg, &dropframe_thresh_arg, argi)) {
       enc_cfg->rc_dropframe_thresh = arg_parse_uint(&arg);
     } else if (arg_match(&arg, &error_resilient_arg, argi)) {
